File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02580/p02580_num1_parsed.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

int main() {
    int H, W, M;
    cin >> H >> W >> M;

    vector<int> row_count(H, 0);
    vector<int> col_count(W, 0);
    unordered_set<long long> targets;

    for (int i = 0; i < M; i++) {
        int h, w;
        cin >> h >> w;
        h--;
        w--;
        row_count[h]++;
        col_count[w]++;
        targets.insert((long long)h * W + w);
    }

    int max_row_count = 0;
    for (int i = 0; i < H; i++) {
        max_row_count = max(max_row_count, row_count[i]);
    }

    int max_col_count = 0;
    for (int i = 0; i < W; i++) {
        max_col_count = max(max_col_count, col_count[i]);
    }

    int max_targets = max_row_count + max_col_count;
    int num_max_row = 0;
    int num_max_col = 0;

    for (int i = 0; i < H; i++) {
        if (row_count[i] == max_row_count) {
            num_max_row++;
        }
    }

    for (int i = 0; i < W; i++) {
        if (col_count[i] == max_col_count) {
            num_max_col++;
        }
    }

    if (num_max_row * num_max_col > M) {
        cout << max_targets << endl;
    } else {
        cout << max_targets - 1 << endl;
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02580/p02580_num1_parsed.py
# Program

H, W, M = map(int, input().split())
targets = []
row_count = [0] * H
col_count = [0] * W

for _ in range(M):
    h, w = map(int, input().split())
    targets.append((h, w))
    row_count[h-1] += 1
    col_count[w-1] += 1

max_targets = max(row_count) + max(col_count)

max_count_rows = sum(1 for i in row_count if i == max(row_count))
max_count_cols = sum(1 for i in col_count if i == max(col_count))
        
for h, w in targets:
    if row_count[h-1] == max(row_count) and col_count[w-1] == max(col_count):
        max_targets -= 1
        
if max_count_rows * max_count_cols > max_targets:
    max_targets = max_count_rows * max_count_cols
        
print(max_targets)